src/js/actions/tools.js

```diff
@@ -36,6 +38,40 @@ define(function (require, exports) {
         policy = require("./policy"),
         shortcuts = require("./shortcuts");
         
+
+    /**
+     * Installs the defaults on a given shape tool
+     * 
+     * @param {string} tool the name of the tool we're using "ellipseTool" or "rectangleTool"
+     * @param {Color} strokeColor a 3 item array represetning the [r,g,b] value of the stroke
+     * @param {number} strokeSize the width of the stroke
+     * @param {number} strokeOpacity the opacity of the stroke
+     * @param {Color} fillColor a 3 item array represetning the [r,g,b] value of the fill
+     *
+     * @return {Promise}
+     */
+    var installShapeDefaultsCommand = function (tool, strokeColor, strokeSize, strokeOpacity, fillColor) {
+        var document = this.flux.store("application").getCurrentDocument(),
+            defaultObj = toolLib.defaultShapeTool(tool, strokeColor, strokeSize, strokeOpacity, fillColor);
+            
+        // If document doesn't exist, or is a flat document
+        if (!document || document.unsupported || document.layers.all.size === 1 &&
+            document.layers.all.first().isBackground) {
+            return descriptor.PlayObject(defaultObj);
+        }
+        var layerSpec = document.layers.allSelected.toList(),
+            layerRef = layerSpec
+                .map(function (layer) {
+                    return layerLib.referenceBy.id(layer.id);
+                })
+                .unshift(documentLib.referenceBy.id(document.id))
+                .toArray();
+
+        var selectObj = layerLib.select(layerRef, false);
+
+        return descriptor.batchPlayObjects([layerLib.deselectAll(), defaultObj, selectObj]);
```

Comment:
Idle comment, in case you have to shove more selection logic here in the near future ... 
since this is basically another flavor of the "selection dance", it might make sense to factor this previous few lines of code into `util/layeractions.js`.